src/clj/puppetlabs/services/config/puppet_server_config_service.clj

```diff
@@ -26,7 +26,9 @@
           "Initializing with the following settings from core Puppet:\n%s"
           (ks/pprint-to-string puppet-config))
         (core/init-webserver! override-webserver-settings!
-                              (get-in tk-config [:webserver])
+                              (get-in tk-config
+                                      [:webserver :puppet-server]
+                                      (get-in tk-config [:webserver]))
                               puppet-config)
```

Comment:
I think this is a pretty straightforward way to accomplish this, which is good.  The only thing that bothers me a little bit is that it's not entirely configuration data-driven.  I had originally been envisioning something like:

1) Get the server that the master-service route from web-routes uses.
2) Get the webserver config for that server.

This would allow you to change the server config in web-routes.conf / webserver.conf without having anything in the code know anything about the server name - or lack thereof - directly.

This approach would be more involved, though, and maybe just the idea of looking up a specific service's route from the web-routes.conf - especially from a "common" service like the config one which doesn't manage its own route(s) - would make the approach of directly coding in knowledge of the server name a better option.

Would be curious what @KevinCorcoran or @cprice404 thinks about this, though.